cupyx/scipy/interpolate/_rgi.py

```diff
@@ -42,47 +42,65 @@ def __init__(self, points, values, method="linear", bounds_error=True,
         self.method = method
         self.bounds_error = bounds_error
 
-        if not hasattr(values, 'ndim'):
-            # allow reasonable duck-typed values
-            values = cp.asarray(values)
-
+        self.grid, self._descending_dimensions = self._check_points(points)
+        self.values = self._check_values(values)
+        self._check_dimensionality(self.grid, self.values)
+        self.fill_value = self._check_fill_value(self.values, fill_value)
+        if self._descending_dimensions:
+            self.values = cp.flip(values, axis=self._descending_dimensions)
+
+    def _check_dimensionality(self, points, values):
         if len(points) > values.ndim:
             raise ValueError("There are %d point arrays, but values has %d "
                              "dimensions" % (len(points), values.ndim))
+        for i, p in enumerate(points):
+            if not cp.asarray(p).ndim == 1:
+                raise ValueError("The points in dimension %d must be "
+                                 "1-dimensional" % i)
+            if not values.shape[i] == len(p):
+                raise ValueError("There are %d points and %d values in "
+                                 "dimension %d" % (len(p), values.shape[i], i))
+
+    def _check_values(self, values):
+        if not hasattr(values, 'ndim'):
+            # allow reasonable duck-typed values
+            values = cp.asarray(values)
```

Comment:
Duck-typed values come back? Also `if hasattr(values, 'dtype')` --- the whole `_check_values` method can be simplified to first call `cp.asarray` then `if not cp.issubdtype(....)` to convert integers into floats (cannot just call asarray(..., dtype=float)) because of complex-valued values.

Reply:
I could remove these lines, but then the code would fail for lists.  Is this expected behavior?  I honestly don't know.  I'm leaning towards "yes", based on the rest of the code, but I was hoping someone would chime in.

Reply:
My suggestion was to always cp.asarray.